components/affix/affix.component.ts

```diff
@@ -81,8 +81,8 @@ export class NzAffixComponent implements AfterViewInit, OnChanges, OnDestroy, On
   private affixStyle?: NgStyleInterface;
   private placeholderStyle?: NgStyleInterface;
   private positionChangeSubscription: Subscription = Subscription.EMPTY;
-  private offsetChanged$ = new ReplaySubject(1);
-  private destroy$ = new Subject<void>();
+  private offsetChanged$ = new ReplaySubject<void>(1);
+  private destroy$ = new Subject<boolean>();
```

Comment:
Why is `destroy$` a `Subject<boolean>` instead of a `Subject<void>`? Can you explain it?

Reply:
it's just a convention, but of course it will work with void. In all the case, in a next PR all the destroy$ Subject will be replaced by the new takeUntilDestroyed operator of Angular

Reply:
> it's just a convention, but of course it will work with void. In all the case, in a next PR all the destroy$ Subject will be replaced by the new takeUntilDestroyed operator of Angular

Can internal logic be refactored using signal in the future?
I think I should be able to do it.

Reply:
> > it's just a convention, but of course it will work with void. In all the case, in a next PR all the destroy$ Subject will be replaced by the new takeUntilDestroyed operator of Angular
> 
> Can internal logic be refactored using signal in the future? I think I should be able to do it.

We did consider signal, but we don't have a complete plan yet, so feel free to submit the codes if you'd like :)

Reply:
> > it's just a convention, but of course it will work with void. In all the case, in a next PR all the destroy$ Subject will be replaced by the new takeUntilDestroyed operator of Angular
> 
> Can internal logic be refactored using signal in the future? I think I should be able to do it.

@MunMunMiao if you are agree I can help you to work on that :)

Reply:
Is it possible to use `takeUntilDestroyed`: https://github.com/angular/angular/blob/main/packages/core/rxjs-interop/src/take_until_destroyed.ts